File: Exercises/101-109 - Small Problems/easy_1/sum_of_digits.rb
# Write a method that takes one argument, a positive integer, and returns the sum of its digits.
# For a challenge, try writing this without any basic looping constructs (while, until, loop, and each).

def sum(number)
  sum = 0

  number.to_s.chars.each { |digit| sum += digit.to_i }
  sum
end

puts sum(23) == 5
puts sum(496) == 19
puts sum(123_456_789) == 45

File: Exercises/210 - Small Problems/list-processing/sum-of-digits.js
/*
Write a method that takes one argument, a positive integer, and returns the sum of its digits. Do this using list processing techniques.

Examples:

sum(23);          // 5
sum(496);         // 19
sum(123456789);   // 45
*/

function sum(num) {
  return String(num).split('').map(Number).reduce(function(sum, digit) {
    return sum + digit;
  });
}

console.log(sum(23));
console.log(sum(496));
console.log(sum(123456789));